### PR TITLE
Add unit tests for uncovered lines

### PR DESCRIPTION
**Pull request by Ricool06**
The testing fairy has visited your repository. 🧚
I noticed `validateAddDailyImageCommand`, `ReadDynamoDbDailyImages.getByDate` had some lines missing unhappy path coverage.
`logEvent` was uncovered but not exported until now, so not sure if there is a better way of covering it without just using it as you do in the actual code.

Anyway, I hope this is helpful.
@SketchingDev

### CODE REVIEW

File: packages/daily-image-api/__tests__/integration/handlerGet.test.ts [hunk 1]
```diff
@@ -113,3 +113,22 @@ describe("Test querying daily image", () => {
...
+  test("Event is logged then next callback is called with event", async () => {
+    const originalConsoleLog = console.log;
+    console.log = jest.fn();
+
+    const expectedResolvedValue = "success";
+    const nextCallback = jest.fn().mockResolvedValue(expectedResolvedValue);
+    const expectedEvent = { test: "event" };
+    const expectedArgs = [1, 2, 3];
+
+    const result = await logEvent(nextCallback)(expectedEvent, ...expectedArgs);
```
Comment (SketchingDev) [inline packages/daily-image-api/__tests__/integration/handlerGet.test.ts]: I've updated the master branch to allow the console to be passed in as a dependency (what you've called `expectedArgs`).

So you can pass the mock logger in this like now:
```
const dependencies: AppDependencies = { logger: mockLogger } as any; // was 'const expectedArgs = [1, 2, 3];'

const result = await logEvent(nextCallback)(expectedEvent, dependencies);
```

File: packages/daily-image-api/__tests__/integration/app/query/storage/ReadDynamoDbDailyImages.test.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import AWS from "aws-sdk";
+import { ReadDynamoDbDailyImages } from "../../../../../app/query/storage/ReadDynamoDbDailyImages";
+
+jest.mock("aws-sdk");
```
Comment (SketchingDev) [inline packages/daily-image-api/__tests__/integration/app/query/storage/ReadDynamoDbDailyImages.test.ts]: Interesting! I would have written the mock to only include the method necessary for the test, but this means you have to assert the type as `any` when passing it into the constructor (which I would always prefer not to have to do) .e.g.

```
const dbMock = {
    scan: jest.fn(() => ({
        promise: jest.fn().mockResolvedValue(null)
      }));
}

const sut = new ReadDynamoDbDailyImages(dbMock as any, "tableName");
```

Comment (Ricool06) [reply packages/daily-image-api/__tests__/integration/app/query/storage/ReadDynamoDbDailyImages.test.ts]: Yes, I did try this, as importing + mocking an entire module seemed a bit heavyweight. However, type safety seemed a reasonable trade, and the test isn't slow either way. :smiley:

File: packages/daily-image-api/__tests__/integration/handlerGet.test.ts [hunk 3]
```diff
@@ -113,3 +113,22 @@ describe("Test querying daily image", () => {
...
+
+describe("logEvent", () => {
+  test("Event is logged then next callback is called with event", async () => {
+    const originalConsoleLog = console.log;
+    console.log = jest.fn();
+
+    const expectedResolvedValue = "success";
+    const nextCallback = jest.fn().mockResolvedValue(expectedResolvedValue);
+    const expectedEvent = { test: "event" };
+    const expectedArgs = [1, 2, 3];
```
[the remark this answers is not in the document]
Comment (Ricool06) [reply packages/daily-image-api/__tests__/integration/handlerGet.test.ts]: I shall get on that now.

[the remark this answers is not in the document]
Comment (Ricool06) [reply packages/daily-image-api/__tests__/integration/handlerGet.test.ts]: @SketchingDev this is done now.

File: packages/daily-image-api/__tests__/integration/app/query/storage/ReadDynamoDbDailyImages.test.ts (hunk 2 above)
Comment (Ricool06) [reply packages/daily-image-api/__tests__/integration/app/query/storage/ReadDynamoDbDailyImages.test.ts]: btw, is this a request to change it to using `as any` or just an observation?

Comment (SketchingDev) [reply packages/daily-image-api/__tests__/integration/app/query/storage/ReadDynamoDbDailyImages.test.ts]: Just an observation

**Pull request merged**